src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -322,26 +376,32 @@ protected void validateSchema(Table table, Schema tableSchema, boolean allowSche
       // validate the missing columns in output schema are nullable fields in BigQuery
       List<String> remainingBQFields = BigQueryUtil.getBqFieldsMinusSchema(bqFields, outputSchemaFields);
       for (String field : remainingBQFields) {
-        if (bqFields.get(field).getMode() != Field.Mode.NULLABLE) {
+        Field.Mode mode = bqFields.get(field).getMode();
+        // Mode is optional. If the mode is unspecified, the column defaults to NULLABLE.
+        if (mode != null && mode != Field.Mode.NULLABLE) {
           collector.addFailure(String.format("Required Column '%s' is not present in the schema.", field),
                                String.format("Add '%s' to the schema.", field));
         }
       }
     }
 
-    // Match output schema field type with BigQuery column type
-    for (Schema.Field field : tableSchema.getFields()) {
-      String fieldName = field.getName();
-      // skip checking schema if field is missing in BigQuery
-      if (!missingBQFields.contains(fieldName)) {
-        ValidationFailure failure = BigQueryUtil.validateFieldSchemaMatches(
-          bqFields.get(field.getName()), field, getConfig().getDataset(), tableName,
-          AbstractBigQuerySinkConfig.SUPPORTED_TYPES, collector);
+    // column type changes should be disallowed if either allowSchemaRelaxation or truncate table are not set.
+    if (!allowSchemaRelaxation || !getConfig().isTruncateTableSet()) {
```

Comment:
So we will allow column type changes even if `update schema` option is not set? If so, please document the behavior.

Reply:
column type changes are only allowed if both allowSchemaRelaxation and truncate table is set otherwise we will run into runtime exception with conflicts between existing data vs new data being added.

Reply:
Makes sense, thanks.